sql/src/main/java/io/crate/metadata/FunctionInfo.java

```diff
@@ -82,6 +84,66 @@ public DataType returnType() {
         return returnType;
     }
 
+
+    /**
+     * If this method returns <code>true</code>, this is a function that
+     * preserves comparison relation truth value for all operators
+     * (for some operators with a little operator mapping).
+     *
+     * while missing an appropriate name for this property,
+     * we call it like this.
+     * <p>
+     * I describes the following attribute:
+     * <p>
+     *
+     * say we have a comparison-query like this:
+     *
+     *  col > 10.5
+     *
+     *  then a function f, that preserves the comparison relation, can be applied so
+     *  that:
+     *
+     *  f(col) > f(10.5)
+     *
+     *  for all col for which col > 10.5 is true.
+     *
+     *  This property holds for the = comparison operator for all functions f.
+     *
+     *  Example 1:
+     *
+     *      if f is defined as f(v) = v + 1
+     *      then col + 1 > 11.5 must be true for all col > 10.5.
+     *      This is indeed true.
+     *
+     *      So f is preserving the comparison relation for >.
+     *      f does this for all comparison operators =, >, <, >=,<=
+     *
+     *  Example 2 (a rounding function):
+     *
+     *      if f is defined as f(v) = ceil(v)
+     *      then ceil(col) > 11 for all col > 10.5.
+     *      But there is 10.8 for which f is 11 and
+     *      11 > 11 is false.
+     *
+     *      Here a simple mapping of the operator will do the trick:
+     *
+     *          > -> >=
+     *          < -> <=
+     *
+     *  Example 3:
+     *
+     *      if f is defined as f(v) = v % 5
+     *      then col % 5 > 0.5 for all col > 10.5
+     *      but there is 20 for which
+     *      f is 0 and
+     *      0 > 0.5 is false.
+     *
+     *      So f does not preserve the comparison relation.
+     */
+    public boolean preservesComparisonRelation() {
```

Comment:
pls find a better name for this as this name is misleading imho. unfortunately I don't have a good suggestion :)